android/libraries/rib-android/src/main/kotlin/com/uber/rib/core/RibActivity.kt

```diff
@@ -41,30 +38,32 @@ import com.uber.rib.core.lifecycle.ActivityLifecycleEvent.Companion.create
 import com.uber.rib.core.lifecycle.ActivityLifecycleEvent.Companion.createOnCreateEvent
 import io.reactivex.CompletableSource
 import io.reactivex.Observable
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.rx2.asObservable
 
 /** Base implementation for all VIP [android.app.Activity]s. */
 abstract class RibActivity : CoreAppCompatActivity(), ActivityStarter, LifecycleScopeProvider<ActivityLifecycleEvent>, RxActivityEvents {
   private var router: ViewRouter<*, *>? = null
-  private val lifecycleBehaviorRelay = BehaviorRelay.create<ActivityLifecycleEvent>()
-  private val lifecycleRelay: Relay<ActivityLifecycleEvent> = lifecycleBehaviorRelay.toSerialized()
-  private val callbacksRelay = PublishRelay.create<ActivityCallbackEvent>().toSerialized()
+  private val lifecycleFlow = MutableSharedFlow<ActivityLifecycleEvent>(1, 0, BufferOverflow.DROP_OLDEST)
```

Comment:
do publish & behavior relays (which act slightly different) both get mapped to the same flow type?

Reply:
In this case yes.

https://github.com/Kotlin/kotlinx.coroutines/issues/2515 is an interesting discussion that sheds some light on this, but the TLDR is that StateFlow is more or less a very opinionated SharedFlow configuration and a simplified API to access the value. As our usecase doesn't meet that opinionated impl (required a default value), we need to go the route of configuring the SharedFlow for the behavior that we need of a BehaviorRelay without default value.